packages/components/Button/src/useButton.ts

```diff
@@ -11,7 +11,7 @@ export const useButton = (props: ButtonProps): ButtonState => {
   // GH #1336: Set focusRef to null if button is disabled to prevent getting keyboard focus.
   const focusRef = disabled ? null : componentRef;
   const onClickWithFocus = useOnPressWithFocus(focusRef, onClick);
-  const pressable = useAsPressable({ ...rest, onPress: onClickWithFocus });
+  const pressable = useAsPressable({ ...rest, disabled, onPress: onClickWithFocus });
```

Comment:
I wonder if we should just pass all the props to it to avoid problems like this in the future, and for other controls? I think this fix is fine though.

Reply:
+1; while we need to setup onPress seemingly the initial props parameters can be passed through in one

Reply:
I updated usePressable to use props in the newest iteration.